### PR TITLE
[docs] Added chapter covering app.js, section on routing config w/ app.js in config chapter, and edited middleware section.

### PR DESCRIPTION
**Pull request by zhouyaoji**

### CODE REVIEW

File: docs/dev_guide/intro/mojito_configuring.rst [hunk 1]
```diff
@@ -1532,6 +1534,280 @@ would call the ``index`` action:
...
+Single Route
+############
+
+**routes.json**
+
+.. code-block:: javascript
+
+   [
+       {
+           "settings": [ "master" ],
```
Comment (caridy) [inline docs/dev_guide/intro/mojito_configuring.rst]: it is important to highlight that routes are not going to computed per request, they will be attach as express routes, and the user of the "settings" in this file are only used in conjuction with the context passed when extending express `mojito.extend(app, {context: {foo: 'bar'}})`.

/cc @imalberto

File: docs/dev_guide/intro/mojito_configuring.rst [hunk 2]
```diff
@@ -1532,6 +1534,280 @@ would call the ``index`` action:
...
+           "regex_path": {
+               "verbs": ["get"],
+               "path": "/:matched_path",
+               "regex": { "matched_path": "\d{1,2}_[Mm]ojitos?" },
```
Comment (caridy) [inline docs/dev_guide/intro/mojito_configuring.rst]: @imalberto is this still supported?

**Comment by caridy** [comment]
few notes from the top of my head:
- `routes.json` is in place just for BC, and we should encourage people to do express routes.
- I think middleware section should be expanded to highlight how to control them at the granular level.
- the `attachRoutes` is only needed if `routes.json` is going to be used.

File: docs/dev_guide/intro/mojito_configuring.rst (hunk 2 above)
Comment (imalberto) [inline docs/dev_guide/intro/mojito_configuring.rst]: @caridy Not anymore. 

@zhouyaoji  This is documented [here](https://github.com/yahoo/mojito/wiki/05.-Mojito-v0.9:-Configuring-Routing#wiki-mojito-v09-4).

**Comment by zhouyaoji** [comment]
@caridy The middleware section in `mojito_appjs.rst` points to the middleware section of `mojito_extensions.rst`, which gives more detail. Is this expansive enough? https://github.com/yahoo/mojito/blob/develop/docs/dev_guide/topics/mojito_extensions.rst#middleware

@caridy @imalberto  Thanks for your feedback. I'm going to create a separate chapter called "Routing" that will cover defining routing paths in `app.js` and add a deprecation note to the `routes.json` documentation in `mojito_configuring.rst`. I'll also note that regular expressions are no longer supported in `routes.json`. I'll try to have this PR updated by tomorrow.

**Comment by caridy** [comment]
yeah, the expand on middleware is excellent.

**Comment by zhouyaoji** [comment]
@imalberto @caridy I added a chapter for defining routing paths in `app.js` called `mojito_routing.rst`, deprecated the `routes.json` section in `mojito_configuring.rst`, and replaced example `routes.json` with equivalent routing paths set in `app.js`. There are a lot of changes, but I am mostly concerned with the new chapter `mojito_routing.rst` and `mojito_configuring.rst`.

**Pull request merged**